File: access_modifiers/access_modifiers.go
package main

import "fmt"

type Character struct {
	Name   string // exported
	health int    // unexported (private to package)
	Stamina int   // exported
}

func NewCharacter(name string) Character {
	return Character{
		Name:    name,
		health:  100,
		Stamina: 50,
	}
}

// Exported method
func (c Character) GetHealth() int {
	return c.health
}

// Unexported method
func (c Character) secretTechnique() {
	fmt.Println("Hidden technique unleashed!")
}

func main() {
	c := NewCharacter("Guts")
	fmt.Println("Name:", c.Name)        // ✅ exported
	fmt.Println("Health:", c.GetHealth()) // ✅ via getter
	// fmt.Println(c.health)           // ❌ can't access directly
}

File: access_modifiers/AccessModifiers.java
class Character {
    public String name = "Default";
    private int health = 100;
    protected int stamina = 50;

    public int getHealth() {
        return health;
    }

    private void secretTechnique() {
        System.out.println("Hidden technique unleashed!");
    }

    protected void rest() {
        System.out.println(name + " regains stamina.");
    }
}

class Warrior extends Character {
    void printStats() {
        System.out.println(name);         // ✅ public
        // System.out.println(health);    // ❌ private
        System.out.println(stamina);      // ✅ protected
        rest();                            // ✅ protected method
    }
}

public class AccessModifiers {
    public static void main(String[] args) {
        Warrior w = new Warrior();
        w.printStats();
        System.out.println("Health: " + w.getHealth());
        // w.secretTechnique(); // ❌ private method not accessible
    }
}
